backend/e4h-services/ingestion-service/app/utils/icc_report_converter.py

```
@@ -0,0 +1,963 @@
+
+def _brand_names_by_asset_type(mdms_client, request_info):
+    """Returns {asset_type_code: {registered brand names}}, e.g. {"PANEL": {"Gautam", "ReNew", ...}}.
+
+    Returns an empty dict (never raises) if the schema can't be fetched - brand validation is then
+    silently skipped, the same graceful-degradation choice already made for
+    facility.FacilitySolarConfigurationRule elsewhere in this codebase, rather than blocking every
+    upload on an MDMS outage.
+    """
+    try:
+        records = mdms_client.fetch_mdms_records(request_info, BRAND_SCHEMA_CODE)
+    except Exception:
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)